solutions/gold/cf-1181D.mdx

```diff
@@ -0,0 +1,104 @@
+---
+id: cf-1181D
+source: CF
+title: Irrigation
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/67727)
+
+## Explanation
+
+We can subtract $n$ from each of the queries, sort the cities by occurrences and queries by year, then answer the queries offline.
+
+Let city $i$ denote the $i$'th smallest city sorted by occurrences, and $occ_i$ represent the number of occurrences. We can loop through each $i$ and track the years that have passed.
+
+Assume cities $1, 2, \dots i$ already have $occ_i$ occurrences. For them to all end up with $occ_{i+1}$ occurrences, we need $i \cdot (occ_{i+1} - occ_i)$ more years, meaning that for the next $i \cdot (occ_{i+1} - occ_i)$ years, cities $1, 2, \dots i$ will be selected one after another.
+
+To answer individual years, we can use a BST to track the initial indices of each city $1, 2, 3 \dots i$. Note that initial indices, not the sorted ones, break the tie for occurrences.
```

Comment:
For this part "we can use a BST to track the initial indices", can you link the module for BST

Reply:
???? this is in the BST module lol???

Reply:
oh shoot I just looked over the writing but ok disregard my comment